pkg/backends/websockets.go

```diff
@@ -47,21 +43,19 @@ func NewWebsocketDialer(address string, tlscfg *tls.Config, extraHeader string,
 func (b *WebsocketDialer) Start(bsf netceptor.BackendSessFunc, errf netceptor.ErrorFunc) {
 	go func() {
 		for {
-			cfg, err := websocket.NewConfig(b.address, b.origin)
-			if err != nil {
-				errf(fmt.Errorf("error creating websocket config"), true)
-				return
-			}
+			dialer := websocket.DefaultDialer
 			if b.tlscfg != nil {
-				cfg.TlsConfig = b.tlscfg
+				dialer.TLSClientConfig  = b.tlscfg
 			}
+			header := make(http.Header, 0)
 			if b.extraHeader != "" {
 				extraHeaderParts := strings.SplitN(b.extraHeader, ":", 2)
 				header := make(http.Header, 0)
 				header.Add(http.CanonicalHeaderKey(extraHeaderParts[0]), extraHeaderParts[1])
-				cfg.Header = header
 			}
-			conn, err := websocket.DialConfig(cfg)
+			header.Add(http.CanonicalHeaderKey("origin"), b.origin)
+			u := url.URL{Scheme: "ws", Host: b.address, Path: "/"}
```

Comment:
Callers need to be able to specify `ws://` or `wss://` - we can't assume one or the other here.  That's why `address` needs to be a full URL.